### PR TITLE
Jitter

### PR DESCRIPTION
**Pull request by aricart**
Added the ability to specify jitter parameters. The jitter is an extra amount of time that is random up to specified value which is added to the standard `reconnectTimeWait`. Two values are exposed `reconnectJitter` and `reconnectJitterTLS`. In addition, there's a `reconnectDelayHandler` where a client can specify a value (in milliseconds) that will be used for delay between reconnects.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -496,6 +499,23 @@ The following is the list of connection options and default values.
...
+
+The intention of the settings is to spread out the number of clients attempting to reconnect to a server over a period of time, 
+and thus preventing a ["Thundering Heard"](https://docs.nats.io/developing-with-nats/reconnect/random).
+
+The relationship between these is:
+
+- If `reconnectDelayHandler` is specified, the client will wait the value returned by this function. No other value will be taken into account.
+- If the client specified TLS options, the client will generate a number between 0 and `reconnectJitterTLS` and add it to
+`reconnectTimeWait`.
+- If the client didn't specify TLS options, the client will generate a number between 0 and `reconnectJItter` and add it to `reconnectTimeWait`.
```
Comment (kozlovic) [inline README.md:516]: case `reconnectJitter` instead of `reconnectJItter`.

File: index.d.ts [hunk 2]
```diff
@@ -78,17 +78,26 @@ export interface ClientOpts {
 	pingInterval?: number,
 	preserveBuffers?: boolean,
 	reconnect?: boolean,
+	// jitter for non-tls connections
+	reconnectJitter?: number,
+	// jitter for tls connections
+	reconnectJitterTLS?: number,
+	// handler for custom tls connection
```
Comment (kozlovic) [inline index.d.ts:85]: not sure about this comment: "// handler for custom tls connection"

File: README.md (hunk 1 above)
Comment (aricart) [reply README.md]: good catch.

File: index.d.ts (hunk 2 above)
Comment (aricart) [reply index.d.ts]: removed these comments as the README file contains the doc for this.

File: README.md (hunk 1 above)
Comment (aricart) [reply README.md]: Fixed

**Pull request merged**